### PR TITLE
2021 04 fix download columns

### PR DESCRIPTION
**Pull request by xwatkins**
## Purpose
Column selection in download was not applied.

## Approach
The list of columns in the state was not the list passed to the url generator.

## Testing
What test(s) did you write to validate and verify your changes?
Added a test to check the url is updated correctly when removing a column.

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

### CODE REVIEW

File: src/shared/components/download/__tests__/Download.spec.tsx [hunk 1]
```diff
@@ -90,18 +95,39 @@ describe('Download component', () => {
...
+    async (value, columnSelect) => {
       const formatSelect = screen.getByTestId('file-format-select');
-      fireEvent.change(formatSelect, { target: { value } });
-      const customise = screen.queryByText('Customize data');
-      if (columnSelect) {
-        expect(customise).toBeInTheDocument();
-      } else {
-        expect(customise).not.toBeInTheDocument();
-      }
+      await act(async () => {
```
Comment (aurel-l) [inline src/shared/components/download/__tests__/Download.spec.tsx:100]: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#wrapping-things-in-act-unnecessarily

File: src/shared/components/download/__tests__/Download.spec.tsx [hunk 2]
```diff
@@ -90,18 +95,37 @@ describe('Download component', () => {
...
+    async (value, columnSelect) => {
       const formatSelect = screen.getByTestId('file-format-select');
       fireEvent.change(formatSelect, { target: { value } });
-      const customise = screen.queryByText('Customize data');
-      if (columnSelect) {
-        expect(customise).toBeInTheDocument();
-      } else {
-        expect(customise).not.toBeInTheDocument();
-      }
+      await waitFor(() => {
```
Comment (aurel-l) [inline src/shared/components/download/__tests__/Download.spec.tsx:101]: Why is this needed? What's the difference with before that now requires this to be wrapped by waitFor?

Comment (xwatkins) [reply src/shared/components/download/__tests__/Download.spec.tsx:101]: The change event triggers a state change, so the elements might not be there at the time of check. I'm not sure it worked as expected before?

Comment (aurel-l) [reply src/shared/components/download/__tests__/Download.spec.tsx:101]: I'm wondering trying to reuse `customise` variable makes it a bit more complex that it needs to be. Seeing that bit of test looks to me like this bit of the article https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-waitfor-to-wait-for-elements-that-can-be-queried-with-find

Comment (xwatkins) [reply src/shared/components/download/__tests__/Download.spec.tsx:101]: So I could switch to `await screen.findByText('Customize data');` and then check for `null` instead of `toBeInTheDocument`.

Comment (xwatkins) [reply src/shared/components/download/__tests__/Download.spec.tsx:101]: Actually it's not great is it? It shouldn't be waiting for something we know shouldn't be there... 🤔

Comment (aurel-l) [reply src/shared/components/download/__tests__/Download.spec.tsx:101]: True. Is there a better question to ask the DOM? Under what condition we are ready to expect that something isn't there not just because it didn't had time to appear, but because it will not be there even after the next render or whatever

Comment (xwatkins) [reply src/shared/components/download/__tests__/Download.spec.tsx:101]: So actually it turns out the async/await wasn't needed so I've reverted it back to how it was.

**Review by aurel-l — APPROVED** [review]
